hibernate-core/src/test/java/org/hibernate/orm/test/mapping/embeddable/strategy/usertype/embedded/Person.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.orm.test.mapping.embeddable.strategy.usertype.embedded;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.hibernate.annotations.CompositeType;
+
+import jakarta.persistence.AttributeOverride;
+import jakarta.persistence.Column;
+import jakarta.persistence.ElementCollection;
+import jakarta.persistence.Embedded;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+
+@Table(name = "people")
+//tag::embeddable-usertype-property[]
+@Entity
+public class Person {
+	@Id
+	public Integer id;
+	@Embedded
+	@AttributeOverride(name = "firstName", column = @Column(name = "first_name"))
+	@AttributeOverride(name = "lastName", column = @Column(name = "last_name"))
+	@CompositeType( NameCompositeUserType.class )
+	public Name name;
+	@ElementCollection
+	@AttributeOverride(name = "firstName", column = @Column(name = "first_name"))
+	@AttributeOverride(name = "lastName", column = @Column(name = "last_name"))
+	@CompositeType( NameCompositeUserType.class )
+	public Set<Name> aliases;
+
+	//end::embeddable-usertype-property[]
+
+	private Person() {
+		// for Hibernate use
+	}
+
+	public Person(Integer id, Name name) {
+		this.id = id;
+		this.name = name;
+	}
+
+	public Integer getId() {
+		return id;
+	}
+
+	public Name getName() {
+		return name;
+	}
+
+	public void setName(Name name) {
+		this.name = name;
+	}
+
+	public void setId(Integer id) {
+		this.id = id;
+	}
+
+	public Set<Name> getAliases() {
+		return aliases;
+	}
+
+	public void setAliases(Set<Name> aliases) {
+		this.aliases = aliases;
+	}
+
+	public void addAlias(Name alias) {
+		if ( aliases == null ) {
+			aliases = new HashSet<>();
+		}
+		aliases.add( alias );
```

Comment:
TBH, I don't think this is a role model for two reasons:
* such lazy initialization is prey of the common race condition issue and a complex "lock doubl check" (https://en.wikipedia.org/wiki/Double-checked_locking) fix would be needed;
* I believe the common collection data structure has its own internal storage lazy initialization feature embedded already; the error-prone and tedious boilerplate might not be worthwhile;
IMHO.